File: src/test/java/de/lukaseichler/recurrentneuralnetwork/datatransformation/ScaleTest.java
package de.lukaseichler.recurrentneuralnetwork.datatransformation;

import java.util.Arrays;
import java.util.Comparator;
import org.testng.annotations.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ScaleTest {

    @Test
    public void scaleListOfInputs() throws Exception {
        assertThat(Scale.scale(Arrays.asList(0.0, 5.0, 10.0)))
                .hasSize(3)
                .containsOnly(0.0, 0.5, 1.0);
    }

    @Test
    public void sortResults() throws Exception {
        assertThat(Scale.scale(Arrays.asList(5.0, 0.0, 10.0)))
                .isSortedAccordingTo(Comparator.<Double>naturalOrder());
    }
}

File: src/test/scala/de/lukaseichler/recurrentneuralnetwork/datatransformation/ScaleTest.scala
package de.lukaseichler.recurrentneuralnetwork.datatransformation

import org.scalatest.Matchers
import org.scalatest.testng.TestNGSuite
import org.testng.annotations.Test

/**
 * @author leichler
 */
class ScaleTest extends TestNGSuite with Matchers {

    @Test
    def scaleListOfInputs() {
      val result = Scale.scale(List(0.0, 5.0, 10.0))
      result should (contain(0.0) and contain(0.5) and contain(1))
    }

  @Test
  def sortResults() {
    val result = Scale.scale(List(0.0, 5.0, 10.0))
    result shouldBe sorted
  }
}